### DISCUSSION TITLE
What if the AI responds "API Error. See console logs for details."?

### DISCUSSION THREAD

**Discussion by xionghaohuai** [brianpetro/obsidian-smart-connections#134] [Q&A]
Why is this phrase prompted after using it a few times? But create a new dialog box and you can continue to use it.

**Comment by brianpetro** [comment]
This is the intended outcome when, for whatever reason, there is an API error during the chat.

By printing this response, you become aware of issues with the API.

In the future, there will be a retry button.

Does that clear things up?

🌴

**Reply by Dexmach5723** [reply]
No. Where is the log and how to troubleshoot. Mine has never worked and I only get this error.

**Reply by Dexmach5723** [reply]
And the API test is successful

**Comment by wo0jun** [comment]
I have the same problem too.

**Reply by brianpetro** [reply]
See this and let me know if it helps https://github.com/brianpetro/obsidian-smart-connections/discussions/134#discussioncomment-6180493

THanks
Brian 🌴

**Comment by tangyuan233** [comment]
Same here

**Reply by brianpetro** [reply]
See this and let me know if it helps https://github.com/brianpetro/obsidian-smart-connections/discussions/134#discussioncomment-6180493

Thanks
Brian 🌴

**Comment by brianpetro** [comment]
@jungnerd @tangyuan233 I think these errors were likely due to an ineffective token calculation, so I just updated Smart Connections to better handle that in version `1.6.28`.

Let me know if that helps!

Thanks for your help in solving this
Brian 🌴

**Reply by tangyuan233** [reply]
1.6.28 works! Thanks!!

**Reply by wo0jun** [reply]
Great! Thanks!

**Comment by kampiamodikuesto** [comment]
With last version i get same error token problem. I try with free openAI account and with pro openAI account but results are the same:

> *API Error. See console logs for details.*

and "Test API" fails.

> Smart Connection: hasn't work as espected!

How can i debug this more deeply?

**Reply by brianpetro** [reply]
Looks like an invalid API key.

You can also check the developer console for errors.

**Reply by kampiamodikuesto** [reply]
yes, console tell me that there is a problem on api. I try with free and pro api from two different chatgpt/openai account. i have generated from here: [https://platform.openai.com/account/api-keys](https://platform.openai.com/account/api-keys)

**Comment by Kmfernan5** [comment]
All I see is API Error. See console logs for details.

https://share.getcloudapp.com/YEu4Yq8K

And when I try to create a new api and enter it and change it from GPT 4 to GPT 3.5 all it says is the following error message see screenshot below. 
https://share.getcloudapp.com/wbuLdjm7

is there a fix?

**Comment by edur0791** [comment]
hey this doesn't seem to work with chatgpt-4-limitedaccess(8k), is this a known issue? I'd like to be able to use ChatGPT-4 with Obsidian's Smart Community plugin, as it is way more accurate.

**Comment by ot4x** [comment]
Hi @brianpetro,

After a couple tries the API keys are not able to verify trough testing, they're also not used when I look trough OpenAI/Keys. My account is in great standing. Seems like a common problem. Have you addressed it somewhere else? 

Looking forward to hearing from you,

Regards

**Comment by Lawrence-of-AnKing** [comment]
I get the same error: API Error. See console logs for details.

The console:

![image](https://github.com/brianpetro/obsidian-smart-connections/assets/22313057/92ad08c5-42bf-4051-950f-4a607a07cb54)

The google gemini token I use works well on chatbox.

**Comment by jason-ruis** [comment]
I'm recently receiving a similar error anytime I try to use a note as context. If I just chat with the Smart Chat like I would any other AI, it works fine. However, when I try to use a note as context, I get this error as well. 
Here's what the Console Log shows as an error:
![Screenshot 2024-06-27 at 3 05 39 PM](https://github.com/brianpetro/obsidian-smart-connections/assets/174059927/51a718be-6545-4dda-8fe3-8d7b93ad9c9c)

**Reply by brianpetro** [reply]
Hi @jason-ruis 

Is it with Gemini only? Or other platforms too?

Also, please confirm the version you are currently using.

And turning on the "Debug at startup" setting in the Obsidian community plugin settings will improve the logs so I can get a better understanding of what's happening.

Thanks for your help in bringing this to my attention and helping to solve the issue 😊
🌴

**Reply by brianpetro** [reply]
update https://github.com/brianpetro/obsidian-smart-connections/issues/675#issuecomment-2198218092
